prompt_toolkit/output/vt100.py

```diff
@@ -455,13 +456,17 @@ def from_pty(
         # Normally, this requires a real TTY device, but people instantiate
         # this class often during unit tests as well. For convenience, we print
         # an error message, use standard dimensions, and go on.
-        fd = stdout.fileno()
+        try:
+            fd = stdout.fileno()
+        except io.UnsupportedOperation as e:
+            fd = -1
 
-        if not stdout.isatty() and fd not in cls._fds_not_a_terminal:
+        if not stdout.isatty() and (fd == -1 or fd not in cls._fds_not_a_terminal):
```

Comment:
Thanks for this!

One remark here:
- Rather than using -1, can we use ``None``, that way the type becomes ``Optional[int]``, and Mypy will guarantee that it won't be added to a list of integers.

Reply:
Right! fixed so.